library/public/interactor.h

```diff
@@ -65,77 +116,89 @@ class F3D_EXPORT interactor
   ///@}
 
   ///@{ @name Bindings
-  /**
-   * Enumeration of supported modifier combination, in binary.
-   */
-  enum class ModifierKeys : unsigned char
-  {
-    ANY = 0x80,      // 10000000
-    NONE = 0x0,      // 00000000
-    CTRL = 0x1,      // 00000001
-    SHIFT = 0x2,     // 00000010
-    CTRL_SHIFT = 0x3 // 00000011
-  };
-
   /**
    * Remove all existing interaction commands and add all default bindings
    * see INTERACTIONS.md for details.
    */
   virtual interactor& initBindings() = 0;
 
   /**
-   * Use this method to add binding, in order to trigger commands for a specified interaction and
-   * modifiers flag.
+   * Use this method to add binding, in order to trigger commands for a specific bind
    *
-   * interaction can be a pressed key symbol, eg: "C",
+   * Bind modifiers is a binary flag from the dedicated enum that represent KeyModifiers.
+   * Bind interaction can be a pressed key symbol, eg: "C",
    * or a dedicated key symbol for special keys:
    * "Left", "Right", "Up", "Down", "Space", "Enter", "Escape", "Question".
    *
-   * modifiers is a binary flag from the dedicated enum that represent KeyModifiers.
+   * group is an optional arg to group bindings together for better display of the documentation.
+   * Groups are kept in order of addition when recovered using `getBindGroups`.
+   * Bindings are kept in order of addition when recovered using `getBindsForGroup`.
    *
-   * When the corresponding interaction and modifiers happens, the provided commands will be
-   * triggered using triggerCommand.
-   * Considering checking if an interaction exists or removing it before adding it to avoid
-   * potential conflicts.
+   * documentationCallback is an optional function that returns a pair of string,
+   * the first is the doc itself, the second is the current value as a string, if any.
+   * Use `getBindingDocumentation` to access this doc.
+   *
+   * When the corresponding bind happens, the provided commands will be triggered using
+   * triggerCommand. Considering checking if an interaction exists or removing it before adding it
+   * to avoid potential conflicts.
    *
    * ANY modifier interactions will only be triggered if no other interaction bind with modifier
    * is found.
    *
-   * Adding commands for an existing combination of interaction and modifier will throw a
-   * interactor::already_exists_exception.
+   * Adding commands for an existing bind will throw a interactor::already_exists_exception.
    */
-  virtual interactor& addBinding(
-    const std::string& interaction, ModifierKeys modifiers, std::vector<std::string> commands) = 0;
+  virtual interactor& addBinding(const interaction_bind_t& bind, std::vector<std::string> commands,
+    std::string group = std::string(),
+    std::function<std::pair<std::string, std::string>()> documentationCallback = nullptr) = 0;
```

Comment:
don't you have an alias for that to improve readability?

Reply:
Not in the public header, I dont think this would be a good idea.

Reply:
why?

Reply:
because then the name  would need to be f3d_documentation_callback_t or something.

Reply:
Idk, since it's just an alias for convenience, I don't see it as a true type. We have an example here: https://github.com/f3d-app/f3d/blob/d8ea37a730ba8ca7f5373c8337111ad463917874/library/public/context.h#L26

Reply:
You are not afraid of conflicts with other headers ?

Reply:
not if the alias is in the class scope

Reply:
🤦